### ISSUE TITLE
Async support for CacheLoader / AdvancedCacheLoader

### ISSUE THREAD

**Issue by cruftex** [cache2k/cache2k#93]
General useful for reactive programming.

Useful together with refresh ahead. See: https://github.com/ben-manes/caffeine/issues/261

**Comment by cruftex** [comment]
Shipped as preview in 1.3.2.Alpha and looks good. Tested it in a small internal project and looks good.

Here is an example for using the async Java 11 http client with cache2k:

````
public class GitHubStatisticsJava11Source implements AsyncCacheLoader<String, GitHubCounters> {

  private HttpClient client = HttpClient.newHttpClient();

  public void load(String key, Context<String, GitHubCounters> context, final Callback<GitHubCounters> callback) {
      request(key).whenComplete((v, t) -> {
        if (t != null)
          callback.onLoadFailure(t);
        else
          callback.onLoadSuccess(v);
      });
  }

  public CompletableFuture<GitHubCounters> request(String key) {
    HttpRequest request = HttpRequest.newBuilder()
      .uri(URI.create("https://api.github.com/repos/" + key))
      .build();
    return client.sendAsync(request, HttpResponse.BodyHandlers.ofInputStream())
      .thenApply(HttpResponse::body)
      .thenApply(GitHubStatisticsSource::parse);
  }

}
````

See the project at: https://github.com/sensepitch/badges/blob/master/src/main/java/com/headissue/badges/GitHubStatisticsJava11Source.java

For cache2k 2.0, which can use Java 8 `CompletableFuture` I intend keep the callback based interface but
add another one so, `CompletableFuture` can be used directly, thus avoiding boilerplate code like the above that
converts between the different async interfaces.

**Comment by cruftex** [comment]
Closing, since coding complete. Unfortunately the commit failed to reference this issue.
